Roledeps

cvaraldi opened the pull request:
##### ISSUE TYPE
<!--- Pick one below and delete the rest: -->
 - Feature Pull Request

##### COMPONENT NAME
<!--- Name of the module/plugin/module/task -->
role/metadata

##### ANSIBLE VERSION
<!--- Paste verbatim output from “ansible --version” between quotes below -->
```
ansible-playbook 2.3.0 (roledep cae224d3d2) last updated 2017/01/22 17:31:48 (GMT +200)
  config file = /etc/ansible/ansible.cfg
  configured module search path = Default w/o overrides
```

##### SUMMARY
<!--- Describe the change, including rationale and design decisions -->

<!---
If you are fixing an existing issue, please include "Fixes #nnn" in your
commit message and your description; but you should still explain what
the change does.
-->

<!-- Paste verbatim command output below, e.g. before and after your change -->
```
add new ROLE_SKIP_DEPENDENCIES  configuration variable (boolean, default false):
if false: act normally
if true: don't load dependencies from metadata.
this can be useful when we have two roles A and B in a playbook.
Let's say:
- hosts: localhost
  roles:
    - { role: yum, tags: ["yum","dnf"] }
    - { role: sshd, tags: ["sshd"] }
sshd depends on yum (in metadata/main.yml), but you occasionnally want to run sshd without running tasks from yum role (because it would update all packages), and you don't even want to run tasks in yum.
This of course implies tag "sshd" (in the given example) is only in sshd role.
In case of complex playbooks, this might accelerate the run drastically if one want to run a single role/tag.
```

mwedgwood-rmn committed:
Add account ID grouping and attribute to ease multi-account management

cvaraldi committed:
add variables that gives the possibility to completely skip dependency resolution.

this can be usefull when A depends on B, A has been updated and needs to be run on targets, B has been updated but we don't want to run it, not even skipping all the tasks within it.

cvaraldi committed:
fix to support new parameters from get_config. Use boolean type

resmo commented on lib/ansible/constants.py:
```diff
@@ -236,6 +236,7 @@ def load_config_file():
 DEFAULT_INTERNAL_POLL_INTERVAL = get_config(p, DEFAULTS, 'internal_poll_interval', None, 0.001, value_type='float')
 ERROR_ON_MISSING_HANDLER  = get_config(p, DEFAULTS, 'error_on_missing_handler', 'ANSIBLE_ERROR_ON_MISSING_HANDLER', True, value_type='boolean')
 SHOW_CUSTOM_STATS = get_config(p, DEFAULTS, 'show_custom_stats', 'ANSIBLE_SHOW_CUSTOM_STATS', False, value_type='boolean')
+DEFAULT_ROLE_SKIP_DEPENDENCIES    = get_config(p, DEFAULTS, 'role_skip_dependencies', 'ROLE_SKIP_DEPENDENCIES', False, value_type='boolean')
```
I would change the naming a bit:

~~~ python
DEFAULT_SKIP_ROLE_DEPENDENCIES    = get_config(p, DEFAULTS, 'skip_role_dependencies', 'ANSIBLE_SKIP_ROLE_DEPENDENCIES', False, value_type='boolean')

~~~

cvaraldi replied:
sounds good to me. Which would need to update lib/ansible/playbook/role/__init__.py too

resmo commented:
nice and useful addition

machacekondra committed:
Add oVirt modules notes to CHANGELOG (#20401)

cvaraldi committed:
rename DEFAULT_ROLE_SKIP_DEPENDENCIES to DEFAULT_SKIP_ROLE_DEPENDENCIES

bcoca committed:
updated timezone docs to avoid repainting bikeshed

fixes #19272
fixes #20570

bcoca committed:
actually fix tickets in previous commit

both modules had similar issues and i conflated them

bcoca committed:
added formatting

dbarrosop committed:
First release of the callback selective.py (#17309)

tntC4stl3 committed:
Support option none in docker_container when setting log_driver (#19040)

Per official docker document, it support setting `--log-driver=none` to
disable any logging for the container. So let's add it to this module.

Fixes #5337

Qalthos committed:
vyos_command 2.3 (#18994)

* Peter's new module_utils/vyos
* Update vyos_command
* Restore `set terminal length $ANSIBLE_VYOS_TERMINAL_LENGTH` in vyos
* vyos_command tests
* Remove provider mentions from vyos_command
* Extend get_config to take (and cache) various `show configuration` commands

cvaraldi commented:
hmmm, I think I made a few mistakes in my git actions trying to rebase my local branch :( Still getting new to git, it's a bit confusing to me :-(

cvaraldi commented:
>> will close this pull request and create a new one
